packages/features/ee/workflows/components/WorkflowStepContainer.tsx

```diff
@@ -578,7 +684,7 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                           ? false
                           : true;
 
-                      if (isPhoneNumberNeeded && isNumberValid && !isEmpty) {
+                      if (isPhoneNumberNeeded && isNumberValid && !isEmpty && numberVerified) {
```

Comment:
Don't open dialog is phone number is not yet verified (error message is set above)